### ISSUE TITLE
Moving average for d_hat

### ISSUE THREAD

**Issue by rahulvigneswaran** [KaihuaTang/Long-Tailed-Recognition.pytorch#28]
In the code the moving average is like this, 
https://github.com/KaihuaTang/Long-Tailed-Recognition.pytorch/blob/90c8b2c0b66d17f78b67263861bc9d858fe20128/classification/run_networks.py#L215

In the above scenario, more importance is being given to newer epochs, in that case, why can't we just use the final epoch's model?
Or is there any other rationale behind it?

Should the line have been like the following instead ?
`self.embed_mean = self.mu * self.embed_mean + (1-self.mu)*self.features.detach().mean(0).view(-1).cpu().numpy()`

where the `self.mu = 0.9`.

**Comment by KaihuaTang** [comment]
Yes, directly calculating the moving average on the last few epochs should be the same. The reason I keep accumulating the d_hat from the beginning is only for the simplicity of the code.
